### PR TITLE
CQRS - Single Query and Command handler registrations supported

### PR DESCRIPTION
**Pull request by Karol-Pawlowski-Allegro**
- Query dispatcher will throw "MultipleQueryHandlerException" if more than single handler registration is made
- Command dispatcher will throw "MultipleCommandHandlerException" if more than single handler registration is made

### CODE REVIEW

**Review by darthkurak — APPROVED** [review]

File: src/.editorconfig [hunk 1]
```diff
@@ -180,6 +180,8 @@ dotnet_diagnostic.CA1012.severity = warning
 # CA1822: Make member static
 dotnet_diagnostic.CA1822.severity = warning
 
+dotnet_diagnostic.CA1852.severity = suggestion
```
Comment (ppiwow-apay) [inline src/.editorconfig:183]: why?

File: src/Allegro.Extensions.Cqrs/version.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <Project>
   <PropertyGroup>
-    <VersionPrefix>2.0.0</VersionPrefix>
+    <VersionPrefix>2.0.1</VersionPrefix>
```
Comment (ppiwow-apay) [inline src/Allegro.Extensions.Cqrs/version.xml:3]: i would increase minor, this is braking change a little bit

File: src/.editorconfig (hunk 1 above)
Comment (Karol-Pawlowski-Allegro) [reply src/.editorconfig:183]: CA1852 is recommendation for sealing the classes and in our code - with tons suggestions. The criticality of it is minor but having warning as errors causes build failing.

File: src/Allegro.Extensions.Cqrs/version.xml (hunk 2 above)
Comment (Karol-Pawlowski-Allegro) [reply src/Allegro.Extensions.Cqrs/version.xml]: you're right, minor version will be more adequate. Fixed

File: src/.editorconfig (hunk 1 above)
Comment (Karol-Pawlowski-Allegro) [reply src/.editorconfig]: I have reverted this change.

**Review by ppiwow-apay — APPROVED** [review]

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Demo/Allegro.Extensions.Cqrs.Demo.csproj [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
```
[the remark this answers is not in the document]
Comment (Karol-Pawlowski-Allegro) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Demo/Allegro.Extensions.Cqrs.Demo.csproj:3]: mistake, fixed.

**Pull request merged**